selfservice/hook/verification.go

```diff
@@ -63,15 +65,23 @@ func (e *Verifier) do(r *http.Request, i *identity.Identity, f flow.Flow) error
 		if address.Status != identity.VerifiableAddressStatusPending {
 			continue
 		}
+		var csrf string
+		if f.GetType() == flow.TypeBrowser {
+			csrf = e.r.CSRFHandler().RegenerateToken(w, r)
+		}
 		verificationFlow, err := verification.NewPostHookFlow(e.r.Config(),
 			e.r.Config().SelfServiceFlowVerificationRequestLifespan(r.Context()),
-			e.r.GenerateCSRFToken(r), r, strategy, f)
+			csrf, r, strategy, f)
 		if err != nil {
 			return err
 		}
 
 		verificationFlow.State = verification.StateEmailSent
 
+		if err := strategy.PopulateVerificationMethod(r, verificationFlow); err != nil {
```

Comment:
NewPostHookFlow doing this already, why do we need to do it again?

Reply:
Because the state changed two lines above this and `PopulateVerificationMethod` uses that to decide whether to render the code or email input. Ideally, `NewPostHookFlow` would set that when creating the flow, but `verification.NewFlow` does not allow this, and changing that would require touching quite a few places, which I didn't want to do in this PR. 

I'd like to follow up here, and maybe refactor that part in a future PR, to use a single payload struct as parameter, as that function already takes in a bunch of parameters, making its use quite difficult, IMO. WDYT?

Reply:
ok